crates/zkforge/bin/cmd/script/build.rs

```diff
@@ -217,57 +229,36 @@ impl ScriptArgs {
-            let output = compile::compile_target_with_filter(
-                &target_contract,
-                &project,
-                self.opts.args.silent,
-                self.verify,
-                filters,
-            )?;
-            return Ok((project, output))
-        }
+        let zksolc_manager = setup_zksolc_manager(DEFAULT_ZKSOLC_VERSION.to_owned())?;
+        let zksolc_opts = ZkSolcOpts {
+            compiler_path: zksolc_manager.get_full_compiler_path(),
+            is_system: false,
+            force_evmla: false,
+        };
 
-        if !project.paths.has_input_files() {
-            eyre::bail!("The project doesn't have any input files. Make sure the `script` directory is configured properly in foundry.toml. Otherwise, provide the path to the file.")
-        }
+        let mut zksolc = ZkSolc::new(zksolc_opts, project);
+        let (output, _) = match zksolc.compile() {
+            Ok(compiled) => compiled,
+            Err(e) => return Err(eyre::eyre!("Failed to compile scripts with zksolc: {}", e)),
+        };
 
         let contract = ContractInfo::from_str(&self.path)?;
         self.target_contract = Some(contract.name.clone());
 
-        // We received `contract_path:contract_name`
-        if let Some(path) = contract.path {
-            let path =
-                dunce::canonicalize(path).wrap_err("Could not canonicalize the target path")?;
-            let output = compile::compile_target_with_filter(
-                &path,
-                &project,
-                self.opts.args.silent,
-                self.verify,
-                filters,
-            )?;
-            self.path = path.to_string_lossy().to_string();
-            return Ok((project, output))
-        }
-
-        // We received `contract_name`, and need to find its file path.
-        let output = if self.opts.args.silent {
-            compile::suppress_compile(&project)
-        } else {
-            compile::compile(&project, false, false)
-        }?;
-        let cache =
-            SolFilesCache::read_joined(&project.paths).wrap_err("Could not open compiler cache")?;
+        // // We received `contract_name`, and need to find its file path.
+        // let output = if self.opts.args.silent {
+        //     compile::suppress_compile(&project)
+        // } else {
+        //     compile::compile(&project, false, false)
+        // }?;
+        // let cache =
+        //     SolFilesCache::read_joined(&project.paths).wrap_err("Could not open compiler cache")?;
 
-        let (path, _) = get_cached_entry_by_name(&cache, &contract.name)
-            .wrap_err("Could not find target contract in cache")?;
-        self.path = path.to_string_lossy().to_string();
+        // let (path, _) = get_cached_entry_by_name(&cache, &contract.name)
+        //     .wrap_err("Could not find target contract in cache")?;
+        // self.path = path.to_string_lossy().to_string();
```

Comment:
Left this comment block in here... because there's still a chance we'll need to implement it for this PR